modules/core/src/main/scala/gem/Location.scala

```diff
@@ -96,11 +92,13 @@ object Location {
     fromFoldable(is.toList)
 
   def fromFoldable[F[_]: Foldable](fi: F[Int]): Location =
-    fi.foldRight(IList.empty[Int]) { (i, lst) =>
-      (lst.isEmpty && i === Int.MinValue) ? lst | i +: lst
-    } match {
-      case ICons(h, t) => new Middle(NonEmptyList.nel(h, t)) {}
-      case _           => Beginning
+    fi.foldRight(Eval.now(List.empty[Int])) { (i, elst) =>
+      elst.map { lst =>
+        if (lst.isEmpty && i === Int.MinValue) lst else (i :: lst)
+      }
+    }.value match {
```

Comment:
Not sure what you mean, sorry. `Foldable.foldRight` wants an `Eval` as the accumulator. Cats does this rather than use by-name arguments. Can this be simplified?